forum/qa-plugin/report-reason/script.js

```diff
@@ -1,32 +1,51 @@
 document.addEventListener("DOMContentLoaded", () => {
     
     const flagboxPopup = document.querySelector("#flagbox-popup");
-    const erorrPopupHtml = document.querySelector('.html-error');
     const flagButton = document.querySelector('.qa-form-light-button-flag');
     flagButton.type = "button";
+    const wrap = document.querySelector('.qa-flag-reasons-wrap');
 
+    function showPopup() {
+        flagboxPopup.classList.remove("hide");
+        flagboxPopup.classList.add("show");
+    }
+    function hidePopup() {
+        flagboxPopup.classList.remove("show");
+        flagboxPopup.classList.add("hide");
+    }
+
+    wrap.addEventListener("click", (e) => {
+
+    e.stopPropagation();
+
+    });
 
     flagButton.addEventListener("click", () => {
+        showPopup();    
+
+        flagboxPopup.addEventListener("click", () => {
+            hidePopup();
```

Comment:
Jeśli do obsługi zdarzenia korzystasz z jednej funkcji i w dodatku bez parametrów, to zamiast przekazywać do listenera anonimową funkcję, przekaż po prostu referencję do tej ostatecznej, czyli w tym przypadku: `flagboxPopup.addEventListener('click', hidePopup);`

Zauważyłem też, że znowu piszesz raz pojedyncze cudzysłowy, a raz podwójne. Trzymamy się **jednej** konwencji i piszemy pojedynczymi.